### ISSUE TITLE
Make hapi a peer dependency

### ISSUE THREAD

**Issue by danielo515** [hapijs/glue#116]
#### Context

* *node version*: 8.9.0
* *glue version*: 5.0.0
* *hapi version*: depends on glue
* *any other relevant information*:

Hello.
I think that hapi should be a peer dependency instead of a normal dependency. It happened to us several times that we have a version of hapi on the package.json (ej, 17.3.1) but instead we're using the one that comes with glue (17.5.0). This could be problematic, specially when possible hapi versions are more broad, for example when it was like `>12 < 13`
Why not specify hapi as a peer dependency ? Having all dependencies you are using on your package.json should be mandatory, and using glue hides the actual hapi version you are using. In fact you can install glue, don't declare hapi at all on your package but still using hapi on your code. This can get very confusing.

Regards

**Comment by WesTyler** [comment]
So, the following is my own personal opinion. Take it for what it is :)

I think that packages that are `require()`d into your library and used as a core piece of functionality should be direct dependencies, as hapi is here.
For me, peer dependencies are things that are meant to work alongside a library; for example, a hapi plugin that exposes a `register` hook and consumes the hapi server object in a callback, but does not actually `require('hapi')` in its source code.

**Comment by WesTyler** [comment]
Also, what version of npm or yarn are you using? Glue uses `17.x.x` as its dependency, so any higher-dependency version of hapi 17 should be used.

Here's an example. I installed and specified hapi 17.0.0 on my test application, and installed glue 5 (containing the hapi 17.x.x dependency). Here's the result of `npm ls hapi`:
```
├─┬ glue@5.0.0
│ └── hapi@17.0.0  deduped
└── hapi@17.0.0
```

**Comment by Marsup** [comment]
Agreed on the non-issue, I don't see how this is confusing, and npm behaves as advertised in glue's readme.

**Comment by Marsup** [comment]
Glue defaults to the latest of the branch it supports, only if you have specific needs you need to add it, it seems to make sense to me. Peer dependency may be useful if glue could be compatible with multiple versions but I guess the current plan is to have major of glue for each major of hapi.

**Comment by csrl** [comment]
See this old version of package.json: https://github.com/hapijs/glue/blame/1aa5d15dfa5e8af6056bc67a18ba035fc88a1f1f/package.json#L21

Currently Glue 5 only supports Hapi 17.  Glue 4 supports hapi 11, 12, 13, 14, 15, or 16, and would default to 16 if no specific version was requested.

For more detail see: https://github.com/hapijs/glue#hapi-version-dependency

As it says, add your specific hapi version dependency along side glue, and it will use that.
